runtime/src/comm/gasnet/comm-gasnet-ex.c

```diff
@@ -686,6 +690,7 @@ int chpl_comm_try_nb_some(chpl_comm_nb_handle_t* h, size_t nhandles)
   return gasnet_try_syncnb_some((gasnet_handle_t*) h, nhandles) == GASNET_OK;
 }
 
+// TODO GEX could be scalable query to gasnet itself
 int chpl_comm_addr_gettable(c_nodeid_t node, void* start, size_t len)
 {
 #ifdef GASNET_SEGMENT_EVERYTHING
```

Comment:
Orthogonal to this PR, but an observation in reading the code below: 

I think this function can make/assert the reasonable assumption that `reqstart <= reqend`, IOW the `len` value is valid and not a ridiculously enormous value that indicates a request that wraps around the top of the address space (if that were violated, then the caller has already gone very wrong before reaching here).

With that assumption, then only two of the comparison operations below are actually required (the first and last). The other two are implied by this property and `segstart < segend` (an invariant guaranteed by GASNet). Further, I believe the C optimizer lacks the invariant knowledge required to automate this optimization, which could save a few instructions per call.

I believe this observation applies equally to both this version of the code using the GASNet-1-style `seginfo_table[]`, and a future GEX version of the code that calls `gex_EP_QueryBoundSegmentNB()` to retrieve the bounds from GASNet.

Reply:
Agreed, I've added a TODO for this to https://github.com/chapel-lang/chapel/issues/14443#issuecomment-1795227749